### PR TITLE
Add option to import/export JSON definitions via copy/paste, and other fixes

### PR DESCRIPTION
**Pull request by lukaskollmer**
# Add option to import/export JSON definitions via copy/paste, and other fixes

## :recycle: Current situation & Problem
- Existing JSON definitions for questionnaires can only be imported by selecting and uploading a file. This can be somewhat annoying if you have the encoded questionnaire stored in some other form.
- Analogously, exporting a questionnaire requires dealing with a JSON file.
- There is a bug where the "slider options" part of the question editing UI incorrectly shows up in some cases (eg when selecting some specific subset of the date entry options).

## :gear: Release Notes 
- Added an "Import JSON" button to the main page, which allows users to paste JSON definitions, instead of having to upload a file (the file upload option still exists unchanged).
- Added a "Copy JSON" button to the JSON preview page, which allows users to copy a questionnaire's JSON definition to the clipboard, instead of having to download it as a file (the file download option still exists unchanged).
- Fixed unrelated parts of the question editing UI showing up sometimes.

## :books: Documentation
n/a

## :white_check_mark: Testing
n/a

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

(Note: i am by no means an experienced React developer, so if there's something i did incorrectly, please point it out and i'll change it.)

### CODE REVIEW

File: src/components/Question/ValidationAnswerTypes/ValidationAnswerTypeString.tsx [hunk 1]
```diff
@@ -22,7 +22,8 @@ const ValidationAnswerTypeString = ({ item }: Props): JSX.Element => {
         {
             display: t('Email'),
             code:
-                ".*@.+",
+                // https://stackoverflow.com/a/201378
+                String.raw`(?:[a-z0-9!#$%&'*+/=?^_\`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_\`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])`,
```
Comment (lukaskollmer) [inline src/components/Question/ValidationAnswerTypes/ValidationAnswerTypeString.tsx:26]: this is the part of this PR i expect to be most likely to get rejected; let me know if you want me to change it back to the more simple one.

Comment (lukaskollmer) [reply src/components/Question/ValidationAnswerTypes/ValidationAnswerTypeString.tsx]: @Supereg thanks for linking the discussion; i changed it back to the simple one.

**Review by vishnuravi — COMMENTED** [review]
Hi @lukaskollmer, thanks for the PR! These are great updates. The 'Paste JSON' option is very convenient. I wonder if it makes sense to also move the file upload option into the new modal you've created such that the user could either paste in JSON or choose a file from the same modal, with a single 'Upload Existing Survey' button on the Home screen launching the modal. What do you think?

**Review by vishnuravi — APPROVED** [review]
That makes sense to me @lukaskollmer!

**Pull request merged**